typespec-extension/src/code-model-builder.ts

```diff
@@ -566,108 +519,146 @@ export class CodeModelBuilder {
         codeModelClient.apiVersions = [];
         for (const version of this.getFilteredApiVersions(
           this.apiVersion,
-          versioning.getVersions(),
+          versions,
           this.options["service-version-exclude-preview"],
         )) {
           const apiVersion = new ApiVersion();
-          apiVersion.version = version.value;
+          apiVersion.version = version;
           codeModelClient.apiVersions.push(apiVersion);
         }
       }
 
-      // server
+      // client initialization
       let baseUri = "{endpoint}";
-      const servers = getServers(this.program, client.service);
-      if (servers && servers.length === 1 && !this.isArmSynthesizedServer(servers[0])) {
-        baseUri = servers[0].url;
-      }
-      const hostParameters = this.processHost(servers?.length === 1 ? servers[0] : undefined);
-      codeModelClient.addGlobalParameters(hostParameters);
+      let hostParameters: Parameter[] = [];
+      client.initialization.properties.forEach((initializationProperty) => {
+        if (initializationProperty.kind === "endpoint") {
+          let sdkPathParameters: SdkPathParameter[] = [];
+          if (initializationProperty.type.kind === "union") {
+            if (initializationProperty.type.values.length <= 2) {
```

Comment:
When would we have a union with values length < 2?

Reply:
I used `<=2` becauase we can't handle multiple endpoint case which is `>2`, and `>2`'s opposite is `<=2`.  But I agree it is not possible in real case to have only one union value.  I can change it to to `===2`.